src/Extensions/Logging/Internal/Replayable/ReplayableActivity.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Omex.Extensions.Logging.Replayable
+{
+	internal class ReplayableActivity : Activity
+	{
+		public ReplayableActivity(string operationName) : base(operationName) =>
+			m_logEvents = new ConcurrentBag<LogMessageInformation>();
+
+
+		public void AddLogEvent(LogMessageInformation logEvent) =>
+			m_logEvents.Add(logEvent);
+
+
+		public IEnumerable<LogMessageInformation> GetLogEvents() =>
+			m_logEvents;
+
+
+		private readonly ConcurrentBag<LogMessageInformation> m_logEvents;
```

Comment:
How are you ensuring order here? I'm guessing re-plays need to have ordered events?

Reply:
I was not ensuring it, hope was that log messages would have some timestamps, but it looks like it's not the case. I'll change `ConcurrentBag` to `ConcurrentQueue` although I'm a bit worried about performance implication, we will see how it will go.